app/models/user.py

```diff
@@ -17,6 +21,40 @@ class User(baseModel):
 
     def __str__(self):
         return str(self.__dict__)
+    
+    def __init__(self,*args, **kwargs):
+        super().__init__(*args,**kwargs)
+        self._ldsCache = None
+    
+    @property
+    def isLaborDepartmentStudent(self):
+        if self._ldsCache is not None:
+            return self._ldsCache
+       
+        if not self.student:
+            self._ldsCache = False
+            return False
```

Comment:
I would move this block above the first one in this function, and then you don't need the return because the next conditional will return the cache value. That way you have one fewer exit from the function, which makes it a little easier to work with.